From: Cyrill Gorcunov <gorcunov@gmail.com>
Date: Sun, 7 Nov 2010 18:01:45 +0000 (+0300)
Subject: ppp: Restore virtualization v3
X-Git-Url: http://git.openvz.org/?p=linux-2.6.32-openvz;a=commitdiff_plain;h=aaf7d79ecd409182c101fa22a977b11fa95122f1

ppp: Restore virtualization v3

net, ppp: ppp_open should check for CAP_VE_NET_ADMIN
http://bugzilla.openvz.org/show_bug.cgi?id=1663

Signed-off-by: Cyrill Gorcunov <gorcunov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---

diff --git a/drivers/net/ppp_generic.c b/drivers/net/ppp_generic.c
index f8545d0..bece346 100644
--- a/drivers/net/ppp_generic.c
+++ b/drivers/net/ppp_generic.c
@@ -369,7 +369,7 @@ static int ppp_open(struct inode *inode, struct file *file)
 	/*
 	 * This could (should?) be enforced by the permissions on /dev/ppp.
 	 */
-	if (!capable(CAP_NET_ADMIN))
+	if (!capable(CAP_VE_NET_ADMIN))
 		return -EPERM;
 	if (!net_generic(get_exec_env()->ve_netns, ppp_net_id)) /* no VE_FEATURE_PPP */
 		return -EACCES;
